src/utils/should-component-update.ts

```diff
@@ -0,0 +1,57 @@
+import { Reducers, State } from '../../default';
+import { ReactNComponent, ReactNPureComponent } from '../../types/component';
+import { ReactNCShouldComponentUpdate } from '../methods';
+
+
+
+/*
+type ShouldComponentUpdate<P extends {} = {}, S extends {} = {}> =
+  (nextProps: P, nextState: S, context: any) => void;
+*/
+
+
+
+// this.shouldComponentUpdate on instance
+/*
+export const shouldComponentUpdateInstance = <
+  P extends {} = {},
+  S extends {} = {},
+  G extends {} = State,
+  R extends {} = Reducers,
+  SS = any,
+>(
+  that: ReactNComponent<P, S, G, R, SS> | ReactNPureComponent<P, S, G, R, SS>,
+): boolean => {
+  if (Object.prototype.hasOwnProperty.call(that, 'shouldComponentUpdate')) {
+    const instanceCwu: ShouldComponentUpdate<P, S> = that.shouldComponentUpdate;
+    that.shouldComponentUpdate = (...args: [ P, S, any ]): void => {
+      ReactNShouldComponentUpdate(that);
+      instanceCwu(...args);
+    };
+    return true;
+  }
+  return false;
+};
+*/
+
+// this.shouldComponentUpdate on prototype
+export const shouldComponentUpdatePrototype = <
+  P extends {} = {},
+  S extends {} = {},
+  G extends {} = State,
+  R extends {} = Reducers,
+  SS = any,
+>(
+  that: ReactNComponent<P, S, G, R, SS> | ReactNPureComponent<P, S, G, R, SS>,
+): boolean => {
+
+  const proto: ReactNComponent | ReactNPureComponent =
+    Object.getPrototypeOf(that);
+  if (Object.prototype.hasOwnProperty.call(proto, 'shouldComponentUpdate')) {
+    that.shouldComponentUpdate = (...args: [ P, S, any ]): boolean => {
+      ReactNCShouldComponentUpdate(that);
+      return proto.shouldComponentUpdate.bind(that)(...args); // Returns outcome of shouldComponentUpdate method
+    };
```

Comment:
It should `return true` here, similar to how [`componentWillUpdate` returns](https://github.com/CharlesStover/reactn/blob/master/src/utils/component-will-update.ts#L61).

This check is used where you call `if (!shouldComponentUpdatePrototype) { that.shouldComponentUpdate = () => {}` in another file in this PR.

This file reads, "If this component DOES have a shouldComponentUpdate method on its prototype, set a custom shouldComponentUpdate method on its instance that clears the global subscriptions and then calls the prototype method."

The other file reads, "If this component DOES NOT have a shouldComponentUpdate method on its prototype (and as a result, does not clear subscriptions on its instance), then create a shouldComponentUpdate method on the instance (and don't call anything on the prototype)."